ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -632,22 +632,36 @@ PrjFS_Result PrjFS_WriteFileContents(
 static Message ParseMessageMemory(const void* messageMemory, uint32_t size)
 {
     const MessageHeader* header = static_cast<const MessageHeader*>(messageMemory);
-    if (header->pathSizeBytes + sizeof(*header) != size)
+    if (size != Message_EncodedSize(header))
     {
-        fprintf(stderr, "ParseMessageMemory: invariant failed, bad message? PathSizeBytes = %u, message size = %u, expecting %zu\n",
-            header->pathSizeBytes, size, header->pathSizeBytes + sizeof(*header));
+        fprintf(stderr, "ParseMessageMemory: invariant failed, bad message? message size = %u, expecting minimum of %zu\n",
+            size, sizeof(*header));
```

Comment:
Can we pass this message to `s_callbacks.LogError` instead of (or in addition to) printing to the console?

I suspect `s_callbacks.LogError` did not exist when the old code was written.